synapse/storage/databases/main/stream.py

```diff
@@ -870,7 +862,11 @@ def _get_max_topological_txn(self, txn: LoggingTransaction, room_id: str) -> int
         )
 
         rows = txn.fetchall()
-        return rows[0][0] if rows else 0
+        # An aggregate function like MAX() will always return one row per group
+        # so we can safely rely on the lookup here. For example, when a we
+        # lookup a `room_id` which does not exist, `rows` will look like
+        # `[(None,)]`
+        return rows[0][0] if rows[0][0] is not None else 0
```

Comment:
Even though this function was annotated to always return `int`, it previously returned `None` when the `room_id` didn't exist in the database (`rows = [(None,)]` in this case).

Now we properly fallback to `0` `topological_ordering` when there is no room entry.